.github/workflows/pr-preview.yml

```diff
@@ -0,0 +1,99 @@
+name: PR Preview
+
+on:
+  push:
+    branches:
+      - '**'
+
+jobs:
+  pr-preview:
+    runs-on: ubuntu-latest
+
+    permissions:
+      actions: read
+
+    defaults:
+      run:
+        shell: bash -ileo pipefail {0}
+
+    steps:
+      - name: Checkout Streamlit code
+        uses: actions/checkout@v3
+        with: 
+          persist-credentials: false
+          submodules: 'recursive'
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+      - name: Setup virtual env
+        uses: ./.github/actions/make_init
+      - name: Create Wheel File
+        timeout-minutes: 120
+        run: |
+          sudo apt install rsync
+          BUILD_AS_FAST_AS_POSSIBLE=1 make package
+      - name: Set PREVIEW_BRANCH & BRANCH env variable
+        uses: ./.github/actions/preview_branch
+        with:
+          pull_request_number: ${{ github.event.pull_request.number }}
+          ref_type: ${{ github.ref_type }}
+          branch: ${{ github.ref_name }}
+
+      # Need the AWS S3 secret/key for the below steps - added the if statement to gate for now
+      - if: ${{ 'true' == 'false' }}
+        name: Upload wheel to S3
+        run: |
+          sudo apt install -y awscli
+
+          aws configure set aws_access_key_id "$CORE_PREVIEWS_S3_KEY_ID"
+          aws configure set aws_secret_access_key "$CORE_PREVIEWS_S3_SECRET_KEY"
+
+          # GitHub upgraded ubuntu-latest environment from 18.04 to 20.04 which upgraded aws-cli
+          # from v1 to v2. When running in GH env, v2 of aws-cli throws an error if no region is specified.
+          aws configure set region us-west-2
+
+          cd lib/dist
+          export WHEELFILE="$(ls -t *.whl | head -n 1)"
+
+          if [ ${BRANCH} = "release/demo" ]
```

Comment:
```suggestion
          if [ "${BRANCH}" = "release/demo" ]
```
> consider using double quote shell variables to avoid [word splitting](https://github.com/koalaman/shellcheck/wiki/SC2086), but this is [one of many](https://mywiki.wooledge.org/BashPitfalls) general recommendations for writing shell scripts, and is not specific to GitHub Actions.

https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#good-practices-for-mitigating-script-injection-attacks